app/forms/move_location_form.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+##
+# This will create a persisted scan.
+# It can be used from a view or elsewhere.
+class MoveLocationForm
+  include ActiveModel::Model
+
+  attr_reader :current_user, :user_code, :controller, :action, :parent_location_barcode, :child_location_barcodes, :parent_location, :child_locations, :params
+
+  validate :check_user, :check_parent_location, :check_child_locations
+
+  def initialize; end
+
+  def form_name
+    :move_location_form
+  end
+
+  def submit(params)
+    @params = params
+    assign_params
+    assign_attributes
+    if valid?
+      parent_location.children = child_locations
```

Comment:
Ok, just a question, so the idea is that will not add new locations to current list, but overwrite with a new list of locations. Is that right?

Reply:
Good sport. I think the method itself just changes the parent on each location. I will add a test to check that it doesn't overwrite.

Reply:
I have just checked and it does just add children.